pkg/certmaker/certmaker.go

```diff
@@ -0,0 +1,464 @@
+		return fmt.Errorf("error writing root certificate: %w", err)
+	}
+
+	var signingCert *x509.Certificate
+	var signingKey crypto.Signer
+
+	if intermediateKeyID != "" {
+		// Create intermediate cert if key ID is provided
+		intermediateTmpl, err := ParseTemplate(intermediateTemplatePath, rootCert)
+		if err != nil {
+			return fmt.Errorf("error parsing intermediate template: %w", err)
+		}
+
+		// Initialize new KMS for intermediate key
+		intermediateConfig := config
+		intermediateConfig.RootKeyID = intermediateKeyID
+		intermediateSV, err := InitKMS(context.Background(), intermediateConfig)
+		if err != nil {
+			return fmt.Errorf("error initializing intermediate KMS: %w", err)
+		}
+
+		intermediatePubKey, err := intermediateSV.PublicKey()
+		if err != nil {
+			return fmt.Errorf("error getting intermediate public key: %w", err)
+		}
+
+		intermediateCert, err := x509util.CreateCertificate(intermediateTmpl, rootCert, intermediatePubKey, signerWrapper{sv})
+		if err != nil {
+			return fmt.Errorf("error creating intermediate certificate: %w", err)
+		}
+
+		if err := WriteCertificateToFile(intermediateCert, intermediateCertPath); err != nil {
+			return fmt.Errorf("error writing intermediate certificate: %w", err)
+		}
+
+		signingCert = intermediateCert
+		signingKey = signerWrapper{intermediateSV}
+	} else {
+		signingCert = rootCert
+		signingKey = signerWrapper{sv}
+	}
+
+	// Create leaf cert
+	leafTmpl, err := ParseTemplate(leafTemplatePath, signingCert)
+	if err != nil {
+		return fmt.Errorf("error parsing leaf template: %w", err)
+	}
+
+	// Initialize new KMS for leaf key
+	leafConfig := config
+	leafConfig.RootKeyID = config.LeafKeyID
+	leafSV, err := InitKMS(context.Background(), leafConfig)
+	if err != nil {
+		return fmt.Errorf("error initializing leaf KMS: %w", err)
+	}
+
+	leafPubKey, err := leafSV.PublicKey()
+	if err != nil {
+		return fmt.Errorf("error getting leaf public key: %w", err)
+	}
+
+	leafCert, err := x509util.CreateCertificate(leafTmpl, signingCert, leafPubKey, signingKey)
+	if err != nil {
+		return fmt.Errorf("error creating leaf certificate: %w", err)
+	}
+
+	if err := WriteCertificateToFile(leafCert, leafCertPath); err != nil {
+		return fmt.Errorf("error writing leaf certificate: %w", err)
+	}
+
+	return nil
+}
+
+// WriteCertificateToFile writes an X.509 certificate to a PEM-encoded file
+func WriteCertificateToFile(cert *x509.Certificate, filename string) error {
+	certPEM := &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: cert.Raw,
+	}
+
+	file, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("failed to create file %s: %w", filename, err)
+	}
+	defer file.Close()
+
+	if err := pem.Encode(file, certPEM); err != nil {
+		return fmt.Errorf("failed to write certificate to file %s: %w", filename, err)
+	}
+
+	// Determine cert type
+	certType := "root"
+	if !cert.IsCA {
+		certType = "leaf"
+	} else if cert.MaxPathLen == 0 {
```

Comment:
I'd check that the SKID and AKID match to check that it's a root, as max path length = 0 could be set for a root as well.